test/e2e/tests/test_notebook_instance_lifecycle.py

```diff
@@ -0,0 +1,131 @@
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["NOTEBOOK_INSTANCE_LFC_NAME"] = notebook_instance_lfc_name
+    reference, spec, resource = create_sagemaker_resource(
+        resource_plural="notebookinstancelifecycleconfigs",
+        resource_name=notebook_instance_lfc_name,
+        spec_file="notebook_instance_lifecycle_config",
+        replacements=replacements,
+    )
+    assert resource is not None
+    yield (reference, resource, spec)
+    if k8s.get_resource_exists(reference):
+        _, deleted = k8s.delete_custom_resource(reference, 10, 5)
+        assert deleted
+
+
+def get_notebook_instance_lifecycle_config(notebook_instance_lfc_name: str):
+    try:
+        resp = sagemaker_client().describe_notebook_instance_lifecycle_config(
+            NotebookInstanceLifecycleConfigName=notebook_instance_lfc_name
+        )
+        return resp
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a Notebook Instance Lifecycle Configuration with the name {notebook_instance_lfc_name}. Error {error}"
+        )
+        return None
+
+
+@service_marker
+@pytest.mark.canary
+class TestNotebookInstanceLifecycleConfig:
+    def wait_until_update(self, reference, current_time):
+        for i in range(10):
+            resource = k8s.get_resource(reference)
+            assert resource is not None
+            assert "lastModifiedTime" in resource["status"]
+            lastModifiedTime = resource["status"]["lastModifiedTime"]
+            d = datetime.datetime.strptime(lastModifiedTime, "%Y-%m-%dT%H:%M:%SZ")
+            if d > current_time:
+                return
+            sleep(1)
+        return
+
+    def test_CreateUpdateDeleteNotebookLifecycleConfig(
+        self, notebook_instance_lifecycleConfig
+    ):
+        (reference, resource, spec) = notebook_instance_lifecycleConfig
+        assert k8s.get_resource_exists(reference)
+
+        # Getting the resource name
+        notebook_instance_lfc_name = resource["spec"].get(
+            "notebookInstanceLifecycleConfigName", None
+        )
+        assert notebook_instance_lfc_name is not None
+        current_time = datetime.datetime.today()
+        sleep(5)  # Done to avoid flakiness
+        # Verifying that its set correctly
+        notebook_instance_lfc_desc = get_notebook_instance_lifecycle_config(
+            notebook_instance_lfc_name
+        )
+        assert (
+            notebook_instance_lfc_desc["OnStart"][0]["Content"]
+            == spec["spec"]["onStart"][0]["content"]
+        )
+        assert (
+            k8s.get_resource_arn(resource)
+            == notebook_instance_lfc_desc["NotebookInstanceLifecycleConfigArn"]
+        )
+        spec["spec"]["onStart"] = [
+            {"content": "cGlwIGluc3RhbGwgc2l4"}
+        ]  # cGlwIGluc3RhbGwgc2l4 = pip install six
+        k8s.patch_custom_resource(reference, spec)
```

Comment:
was it decided to include the update test in the same test ? 
If yes, could you just add a comment to indicate this here.

Reply:
added a comment